substrate/frame/tx-pause/src/lib.rs

```diff
@@ -15,6 +15,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! # Transaction Pause
+//!
+//! Allows dynamic, chain-state-based pausing and unpausing of specific extrinsics via call filters.
+//!
+//! ## Pallet API
+//!
+//! See the [`pallet`] module for more information about the interfaces this pallet exposes,
+//! including its configuration trait, dispatchables, storage items, events and errors.
+//!
+//! ## Overview
+//!
+//! A dynamic call filter that can be controlled with extrinsics.
+//!
+//! Pausing an extrinsic means that the extrinsic CANNOT be called again until it is unpaused.
+//! The exception is calls that use `dispatch_bypass_filter`, typically only with the root origin.
+//!
+//! ### Primary Features
+//!
+//! - Calls that should never be paused can be added to a whitelist.
+//! - Separate origin configuration for pausing and pausing.
+//! - Pausing is triggered using the string representation of the call.
+//! - Pauses can target a single extrinsic or an entire pallet.
+//!
```

Comment:
One niche thing here, not sure if it is worth to mention, but it is possible to pause transactions of not-yet deployed pallets. This could be useful to selectively open up functionality of a pallet after its initial deployment.